client/src/modules/cohort/model/reducer.js

```diff
@@ -96,38 +129,70 @@ const cohorts = (state = {}, action) => {
       return { ...state, [_id]: payload };
     }
     case CohortActionTypes.ADD_MEMBER_SUCCESS:
-    case CohortActionTypes.REMOVE_MEMBER_SUCCESS:
+    case CohortActionTypes.REMOVE_MEMBER_SUCCESS: {
+      const {
+        payload: { cohortId, externalAction }
+      } = action;
+
+      if (state[cohortId]) {
+        const { members } = state[cohortId];
+        const cohort = {
+          ...state[cohortId],
+          members: membersReducer(members, action),
+          externalAction
+        };
+
+        return { ...state, [cohortId]: cohort };
+      }
+
+      return state;
+    }
     case CohortActionTypes.ADD_OWNER_ROLE_SUCCESS: {
       const {
         payload: { cohortId, isCurrentUserRoleChanging }
       } = action;
-      const { members } = state[cohortId];
-      const cohort = {
-        ...state[cohortId],
-        members: membersReducer(members, action)
-      };
 
-      if (isCurrentUserRoleChanging) {
-        cohort.isOwner = true;
+      if (state[cohortId]) {
+        const { members } = state[cohortId];
+        if (members) {
+          const cohort = {
+            ...state[cohortId],
+            members: membersReducer(members, action)
+          };
+
+          if (isCurrentUserRoleChanging) {
```

Comment:
this is strange to me..
is I'm changing the role to anybody else I'm already an owner so why this reducer is setting isOwner to true??